### PR TITLE
oncokb-annotator output goes to github

### PR DESCRIPTION
**Pull request by sheridancbio**
The output of oncokb-annotator is committed to a new github repository rather than being copied to ramen.

### CODE REVIEW

File: import-scripts/oncokb-annotator.sh [hunk 1]
```diff
@@ -311,12 +307,30 @@ FLOCK_FILEPATH="/data/portal-cron/cron-lock/oncokb-annotator.sh"
...
         echo "Running rsync to download.cbioportal.org for OncoKB Annotation for MSKSOLIDHEME"
-        rsync_oncokb_annotated_files
+        rsync --exclude "/data/portal-cron/git-repos/oncokb-annotated-msk-impact/.git" -a $ONCOKB_OUT_DIR/ $ONCOKB_ANNOTATED_GIT_DIR
     else
         message="OncoKB Annotation failed for MSKSOLIDHEME. Check logs for more details."
         send_failure_messages "$message" "MSKSOLIDHEME OncoKB Annotation Failure" "MSKSOLIDHEME OncoKB Annotation"
     fi
 
+    # Create and push a git amended changeset
+    cd $ONCOKB_ANNOTATED_GIT_DIR
```
Comment (averyniceday) [inline import-scripts/oncokb-annotator.sh:317]: make this commit & push dependent on annotation success

File: import-scripts/oncokb-annotator.sh [hunk 2]
```diff
@@ -297,10 +282,21 @@ FLOCK_FILEPATH="/data/portal-cron/cron-lock/oncokb-annotator.sh"
     if [ $ONCOKB_ANNOTATION_SUCCESS -eq 1 ] ; then
         echo $(date)
         echo "Beginning README generation..."
-        $PYTHON3_BINARY $README_SCRIPT -o $ONCOKB_README_FILE -u $ONCOKB_URL
+        $PYTHON3_BINARY $README_SCRIPT -o $ONCOKB_ANNOTATOR_README_FILE
         if [ $? -ne 0 ] ; then
             echo "Failed to generate README, exiting..."
             ONCOKB_ANNOTATION_SUCCESS=0
+        else
+
```
Comment (averyniceday) [inline import-scripts/oncokb-annotator.sh:290]: remove extra line

File: import-scripts/oncokb-annotator.sh (hunk 1 above)
Comment (sheridancbio) [reply import-scripts/oncokb-annotator.sh]: Done - this step was moved into the if-then conditional block. To reduce nesting, the add/commit/push steps were put into a named function call, and all three steps (gzip, rsync, git commit/push) were standardized in format.

File: import-scripts/oncokb-annotator.sh (hunk 2 above)
Comment (sheridancbio) [reply import-scripts/oncokb-annotator.sh]: Done. Other unneeded whitespace was cleaned up too, and a function positional argument which was being passed unnecessarily was removed.

File: import-scripts/oncokb-annotator.sh [hunk 3]
```diff
@@ -297,10 +282,21 @@ FLOCK_FILEPATH="/data/portal-cron/cron-lock/oncokb-annotator.sh"
     if [ $ONCOKB_ANNOTATION_SUCCESS -eq 1 ] ; then
         echo $(date)
         echo "Beginning README generation..."
-        $PYTHON3_BINARY $README_SCRIPT -o $ONCOKB_README_FILE -u $ONCOKB_URL
+        $PYTHON3_BINARY $README_SCRIPT -o $ONCOKB_ANNOTATOR_README_FILE
         if [ $? -ne 0 ] ; then
             echo "Failed to generate README, exiting..."
             ONCOKB_ANNOTATION_SUCCESS=0
+        else
```
[the remark this answers is not in the document]
Comment (sheridancbio) [reply import-scripts/oncokb-annotator.sh:321]: The Readme generation step is done as part of the annotation. If the readme is not successfully generated, the status variable ONCOKB_ANNOTATION_SUCCESS is set to a failed annotation, so no commit would occur.

File: import-scripts/oncokb-annotator.sh [hunk 4]
```diff
@@ -78,6 +53,46 @@ FLOCK_FILEPATH="/data/portal-cron/cron-lock/oncokb-annotator.sh"
...
+            send_failure_messages "$message" "oncokb-annotator failed to run." "$message"
+            exit 1
+        fi
+        if ! git push --force ; then
+            message="Error occurred during 'git push --force'"
+            send_failure_messages "$message" "oncokb-annotator failed to run." "$message"
+            exit 1
+        fi
+        cd $wd
+    fi
```
Comment (averyniceday) [inline import-scripts/oncokb-annotator.sh:94]: looks like theres an extra closing if here, not sure if other code was dropped or missing. Causing unit test to fail.

**Review by averyniceday — CHANGES_REQUESTED** [review]
failing unit tests - looks like there's an extra if, not sure if there's missing code or just the extra line

**Comment by sheridancbio** [comment]
> 

fixed

File: import-scripts/oncokb-annotator.sh (hunk 4 above)
Comment (sheridancbio) [reply import-scripts/oncokb-annotator.sh]: fixed

**Review by averyniceday — APPROVED** [review]

**Pull request merged**